### ISSUE TITLE
Calendar Functionality

### ISSUE THREAD

**Issue by ReconOperator** [klejejs/ha-thermia-heat-pump-integration#90]
Noticed that there is a calendar functionality available in the Thermia Online app. Here you can set different tasks, ex block production of hot water during specific times.
Wold it be possible to add this to the integration? 
![Screenshot_20240902_140159_Thermia Online](https://github.com/user-attachments/assets/ccc01542-eb53-4c66-9236-8f5432eafb28)

**Comment by oskari** [comment]
Would be great if this could somehow be integrated.  "Add calendar event" action or something similiar? The thermia online calendar doesn't support schedules (At least for me it doesn't, even if the heat pump itself does), so having an action which I could call and set EVU mode for certain high-cost hours would be fantastic for implementing spot price controlled heating to these "dumb units".

**Comment by ReconOperator** [comment]
> Would be great if this could somehow be integrated. "Add calendar event" action or something similiar? The thermia online calendar doesn't support schedules (At least for me it doesn't, even if the heat pump itself does), so having an action which I could call and set EVU mode for certain high-cost hours would be fantastic for implementing spot price controlled heating to these "dumb units".

As of now i have done this:
Created a input number Helper (holds the number of hours i want to block)
Created a binary sensor (that uses Nordpool HACS integration)

```
{% set numberOfHours = states('input_number.geothermal_expensive_hour_input') | int  %}
{% set expensive = state_attr('sensor.nordpool_se3_no_vat','raw_today') |sort(attribute='value', reverse=true) %}
  {% for i in range(0, numberOfHours) %}
    {% if(now() > expensive[i].start and now() < expensive[i].end) %}
      True
    {% endif %}
  {% endfor %}
```

Then an automation with the binary sensor as trigger. When the binary sensor turns On (expensive hour is On) I call the action to turn the switch for hot water Off. When the binary sensor turns Off (expensive hour is Off) I call the action to turn the switch for hot water On.

**Comment by ReconOperator** [comment]
@flimofly yes I still do it this way. I don't know about useless but as you say I take into account the energy prices from nordpool
